lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
Why exactly have you chosen 100 and what does growth factor have to do with this at all?